File: Examples_ReactNative/SuperpoweredPlayerWithEffects/ios/SuperpoweredModule.m
//
//  Superpowered.m
//  SuperpoweredPlayerWithEffects
//
//  Created by Banto Balazs on 2023. 08. 19..
//

#import <React/RCTBridgeModule.h>
#import <Foundation/Foundation.h>
#import "UIKit/UIKit.h"
@interface RCT_EXTERN_MODULE(SuperpoweredModule, NSObject)
RCT_EXTERN_METHOD(initSuperpowered)
RCT_EXTERN_METHOD(togglePlayback)
RCT_EXTERN_METHOD(enableFlanger:(BOOL)enable)
@end

File: Examples_ReactNative/SuperpoweredPlayerWithEffects/ios/SuperpoweredModule.swift
//
//  Superpowered.swift
//  SuperpoweredPlayerWithEffects
//
//  Created by Banto Balazs on 2023. 08. 18..
//

import Foundation
@objc(SuperpoweredModule)
class SuperpoweredModule : NSObject {
  let superpowered = SuperpoweredAudio()
  
  @objc static func requiresMainQueueSetup() -> Bool {
    return false
  }
  
  @objc func initSuperpowered() -> Void {
    superpowered.initSuperpowered()
  }
  
  @objc func togglePlayback() -> Void {
    superpowered.togglePlayback()
  }
  
  @objc func enableFlanger(_ enable: Bool) -> Void {
    superpowered.enableFlanger(enable)
  }
}
